File: src/test/java/ConversionTest.java
import amf.client.AMF;
import amf.client.model.document.BaseUnit;
import amf.client.parse.Oas20Parser;
import amf.client.parse.Raml10Parser;
import amf.client.render.Oas20Renderer;
import amf.client.render.Raml10Renderer;
import amf.client.resolve.Oas20Resolver;
import amf.client.resolve.Raml10Resolver;
import amf.core.resolution.pipelines.ResolutionPipeline;
import org.junit.BeforeClass;
import org.junit.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.concurrent.ExecutionException;
import java.util.stream.Collectors;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class ConversionTest {

    @BeforeClass
    public static void setup() throws ExecutionException, InterruptedException {
        AMF.init().get();
    }

    @Test
    public void Raml10ToOas20Conversion() throws ExecutionException, InterruptedException, IOException {
        final Raml10Parser parser = new Raml10Parser();
        final Oas20Resolver resolver = new Oas20Resolver();
        final Oas20Renderer renderer = new Oas20Renderer();

        final BaseUnit ramlApi = parser.parseFileAsync("file://resources/examples/banking-api.raml").get();
        final BaseUnit convertedOasApi = resolver.resolve(ramlApi, ResolutionPipeline.COMPATIBILITY_PIPELINE());

        final String result = renderer.generateString(convertedOasApi).get().trim();
        Path path = Paths.get("resources/expected/converted-banking-api.json");
        String read = Files.readAllLines(path).stream().collect(Collectors.joining(System.lineSeparator()));
        assertEquals(read, result);
    }

    @Test
    public void Oas20ToRaml10Conversion() throws ExecutionException, InterruptedException, IOException {
        final Oas20Parser parser = new Oas20Parser();
        final Raml10Resolver resolver = new Raml10Resolver();
        final Raml10Renderer renderer = new Raml10Renderer();

        final BaseUnit oasApi = parser.parseFileAsync("file://resources/examples/banking-api.json").get();
        final BaseUnit convertedRamlApi = resolver.resolve(oasApi, ResolutionPipeline.COMPATIBILITY_PIPELINE());

        // has amf-specific fields for cross-spec conversion support
        final String result = renderer.generateString(convertedRamlApi).get().trim();
        Path path = Paths.get("resources/expected/converted-banking-api.raml");
        String read = Files.readAllLines(path).stream().collect(Collectors.joining(System.lineSeparator()));
        assertEquals(read, result);
    }
}

File: AMF4/src/test/js/conversion.test.js
const amf = require('amf-client-js');
const pipelines = amf.ResolutionPipeline;
const fs = require('fs')
const path = require('path');

beforeAll(() => {
    amf.AMF.init();

});

test('RAML 1.0 to OAS 2.0', () => {
    const parser = new amf.Raml10Parser();
    const resolver = new amf.Oas20Resolver();
    const renderer = new amf.Oas20Renderer();

    return parser.parseFileAsync("file://src/test/resources/examples/banking-api.raml").then(ramlApi => {
        const convertedOasApi = resolver.resolve(ramlApi, pipelines.COMPATIBILITY_PIPELINE)
        return renderer.generateString(convertedOasApi).then(result => {
            const expectedResult = fs.readFileSync(path.resolve(__dirname, '../../../resources/expected/converted-banking-api.json'), 'utf8')
            expect(result).toEqual(expectedResult);
        });
    })
})


test('OAS 2.0 to RAML 1.0', () => {
    const parser = new amf.Oas20Parser();
    const resolver = new amf.Raml10Resolver();
    const renderer = new amf.Raml10Renderer();

    return parser.parseFileAsync("file://src/test/resources/examples/banking-api.json").then(OasApi => {
        const convertedRamlApi = resolver.resolve(OasApi, pipelines.COMPATIBILITY_PIPELINE)
        return renderer.generateString(convertedRamlApi).then(result => {
            const expectedResult = fs.readFileSync(path.resolve(__dirname, '../../../resources/expected/converted-banking-api.raml'), 'utf8')
            expect(result).toEqual(expectedResult);
        });
    })
})
